hasura/graphql-engine
# cli: add deploy option (close #7349)

### Description

Add `deploy` option in the cli to which performs the following actions:

```
$ hasura metadata apply
$ hasura migrate apply --all-databases
$ hasura metadata reload
```

### Changelog


### Affected components


### Related Issues:
- #7349

### Solution and Design

We are invoking the `Run()` functions of CLI options from `metadata` 
and `migrate` options and performing metadata apply, migration apply and 
metadata reload actions in order.

### Steps to test and verify
Run: `hasura deploy --all-databases` and it should perform:
```
hasura metadata apply
hasura migrate apply --all-databases
hasura metadata reload
```

### Server checklist
#### Catalog upgrade
Does this PR change Hasura Catalog version?

#### Metadata
Does this PR add a new Metadata feature?

#### GraphQL

#### Breaking changes
